tests/setup.ts

```diff
@@ -0,0 +1,34 @@
+import * as dotenv from 'dotenv';
+import { chromium } from '@playwright/test';
+import type { FullConfig, Browser } from '@playwright/test';
+
+dotenv.config();
+
+async function globalSetup(config: FullConfig) {
+    const { storageState } = config.projects[0].use;
+
+    const chromiumBrowser = await chromium.launch();
+    await setupPage(chromiumBrowser, storageState as string)
+}
+
+async function setupPage(browser: Browser, storageState: string) {
+    const page = await browser.newPage();
+    await page.goto(`${process.env.APPLICATION_BASE_URL}/logto/sign-in`);
+    await page.waitForURL(`${process.env.LOGTO_ENDPOINT}/sign-in`, {
+		waitUntil: 'domcontentloaded',
+	});
+
+    await page.getByPlaceholder('Email / Username').fill(process.env.TEST_USER_EMAIL);
+    await page.getByRole('button', { name: 'Sign in' }).click();
+    await page.getByPlaceholder('Password').fill(process.env.TEST_USER_PASSWORD);
+    await page.getByRole('button', { name: 'Continue' }).click();
+    await page.waitForURL(`${process.env.APPLICATION_BASE_URL}/swagger/`, {
+        waitUntil: 'domcontentloaded',
+        timeout: 60000,
+    });
+
+    // End of authentication steps.
+	await page.context().storageState({ path: storageState });
+}
```

Comment:
Move logging in to [auth.setup.ts](https://playwright.dev/docs/auth) instead. Check whether you need to explicitly call it from globalSetup, or if it's automatically invoked.

Where we'll _really_ need auth.setup.ts if/when we want to run one set of tests as a testnet user, and the same set of tests as mainnet user. Which basically means parallel workers, logged in as different test users. https://playwright.dev/docs/auth#multiple-signed-in-roles